docs/hub/dduf.md

```diff
@@ -0,0 +1,179 @@
+...         ("text_encoder/config.json", "path/to/text_encoder/config.json"),
+...         ("text_encoder/model.fp16.safetensors", "path/to/text_encoder/model.fp16.safetensors"),
+...         # ... add more entries here
+...     ]
+... )
+```
+
+`export_entries_as_dduf` works well if you've already saved your model on the disk. But what if you have a model loaded in memory and want to serialize it directly into a DDUF file? `export_entries_as_dduf` lets you do that by providing a Python `generator` that tells how to serialize the data iteratively:
+
+```python
+(...)
+
+# Export state_dicts one by one from a loaded pipeline
+>>> def as_entries(pipe: DiffusionPipeline) -> Generator[Tuple[str, bytes], None, None]:
+...     # Build a generator that yields the entries to add to the DDUF file.
+...     # The first element of the tuple is the filename in the DDUF archive. The second element is the content of the file.
+...     # Entries will be evaluated lazily when the DDUF file is created (only 1 entry is loaded in memory at a time)
+...     yield "vae/config.json", pipe.vae.to_json_string().encode()
+...     yield "vae/diffusion_pytorch_model.safetensors", safetensors.torch.save(pipe.vae.state_dict())
+...     yield "text_encoder/config.json", pipe.text_encoder.config.to_json_string().encode()
+...     yield "text_encoder/model.safetensors", safetensors.torch.save(pipe.text_encoder.state_dict())
+...     # ... add more entries here
+
+>>> export_entries_as_dduf(dduf_path="my-cool-diffusion-model.dduf", entries=as_entries(pipe))
+```
+
+### Loading a DDUF file with Diffusers
+
+Diffusers has a built-in integration for DDUF files. Here is an example on how to load a pipeline from a stored checkpoint on the Hub:
+
+```py
+from diffusers import DiffusionPipeline
+import torch
+
+pipe = DiffusionPipeline.from_pretrained(
+    "DDUF/FLUX.1-dev-DDUF", dduf_file="FLUX.1-dev.dduf", torch_dtype=torch.bfloat16
+).to("cuda")
+image = pipe(
+    "photo a cat holding a sign that says Diffusers", num_inference_steps=50, guidance_scale=3.5
+).images[0]
+image.save("cat.png")
+```
+
+## F.A.Q.
+
+### Why build on top of ZIP?
+
+ZIP provides several advantages:
+- Universally supported file format
+- No additional dependencies for reading
+- Built-in file indexing
+- Wide language support
+
+### Why not use a TAR with a table of contents at the beginning of the archive?
+
+See the explanation in this [comment](https://github.com/huggingface/huggingface_hub/pull/2692#issuecomment-2519863726).
+
+### Why no compression?
+
+- Enables direct memory mapping of large files
+- Ensures consistent and predictable remote file access
+- Prevents CPU overhead during file reading
+- Maintains compatibility with safetensors
+
+### Can I modify a DDUF file?
+
+No. For now, DDUF files are designed to be immutable. To update a model, create a new DDUF file.
+	
+### Which frameworks/apps support DDUFs?
+	
+- [Diffusers](https://github.com/huggingface/diffusers)
```

Comment:
Note: we might have Eric's `diffuse-rs` (candle-based) come in by monday too. So we'll have two supported frameworsk on day-0